### PR TITLE
Luap42/new site btn

### PR DESCRIPTION
**Pull request by luap42**
This PR preserves a previously already merged PR state (#130), which was for some extremely weird reasons removed.

Likely not supposed to be merged, but maybe that fixes it.

### CODE REVIEW

File: app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -61,6 +61,85 @@ def audit_log
...
+    settings.find_by(name: 'ModBadgeCharacter')
+            .update(value: default_settings.find_by(name: 'ModBadgeCharacter').value)
+    settings.find_by(name: 'SEApiClientId')
+            .update(value: default_settings.find_by(name: 'SEApiClientId').value)
+    settings.find_by(name: 'SEApiClientSecret')
+            .update(value: default_settings.find_by(name: 'SEApiClientSecret').value)
+    settings.find_by(name: 'SEApiKey')
+            .update(value: default_settings.find_by(name: 'SEApiKey').value)
+    settings.find_by(name: 'AdministratorContactEmail')
+            .update(value: default_settings.find_by(name: 'AdministratorContactEmail').value)
```
Comment (ArtOfCode-) [inline app/controllers/admin_controller.rb:118]: **issue** (code-clarity): this section is not ideal - think DRY. Can we create a hash instead mapping `primary_color: 'SiteCategoryHeaderDefaultColor'` etc instead and iterate through that to apply each setting - and likewise for the auto-load defaults?

File: app/controllers/admin_controller.rb [hunk 2]
```diff
@@ -61,6 +61,85 @@ def audit_log
...
+      'discussion', 'support', 'feature-request', 'bug',
+      'status-completed', 'status-declined', 'status-review', 'status-planned', 'status-deferred'
+    ]
+    Tag.create(tags.map { |t| { name: t, community_id: @community.id, tag_set: TagSet.meta } })
+
+    # Set Meta tags as required/mod-only
+    meta_category = Category.where(name: 'Meta').last
+    meta_category.required_tags << Tag.where(name: ['discussion', 'support', 'feature-request', 'bug'])
+    meta_category.moderator_tags << Tag.where(name: ['status-completed', 'status-declined', 'status-review',
+                                                     'status-planned', 'status-deferred'])
```
Comment (ArtOfCode-) [inline app/controllers/admin_controller.rb:131]: **issue** (bug): the `Tag` selectors here are missing a tag set scope - at present they're going to add every instance of each of these tags from every community to the new community's required/mod tags.

File: app/controllers/admin_controller.rb [hunk 3]
```diff
@@ -61,6 +61,85 @@ def audit_log
...
+
+    # Clear cache
```
Comment (ArtOfCode-) [inline app/controllers/admin_controller.rb:133]: **suggestion** (non-blocking): unnecessary comment - pretty clear what the following line does already

File: app/controllers/admin_controller.rb [hunk 4]
```diff
@@ -61,6 +61,85 @@ def audit_log
...
+    Rails.cache.clear
+
+    # Audit Log
```
Comment (ArtOfCode-) [inline app/controllers/admin_controller.rb:136]: **suggestion** (non-blocking): unnecessary comment - pretty clear what the following line does already

File: app/views/admin/setup.html.erb [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+<p class="is-lead">Welcome to your new <%= @community.name %> community!</p>
+
+<p>We are almost ready. We just need some more information to complete creation of the site.</p>
+
+<p>Don't worry. You will be able to change all these things in the site settings later.</p>
+
+<%= form_tag setup_save_url do %>
+
+<div class="widget"><div class="widget--body">
```
Comment (ArtOfCode-) [inline app/views/admin/setup.html.erb:11]: **issue** (code-clarity): missing an indent level here after the opening `form_tag`. Also, shouldn't have two opening `<div>`s on the same line - separate lines with a new indent level for the second. Likewise the closing tags, and likewise the rest of this file.

**Review by ArtOfCode- — CHANGES_REQUESTED** [review]

File: app/views/admin/setup.html.erb (hunk 5 above)
Comment (galegosimpatico) [reply app/views/admin/setup.html.erb:11]: [https://thecodinglove.com/content/041/cfkRtbl.webm](https://thecodinglove.com/when-i-find-the-missing-div)

**Pull request merged**